src/test/ui/issue-47706-trait.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+trait T {
+    fn f(&self, _: ()) {
+        None::<()>.map(Self::f);
+    }
+    //~^^ ERROR function is expected to take a single 0-tuple as argument,
+    //~^^ but it takes 2 distinct arguments [E0593]
```

Comment:
Just remove line 16, line 15 will match the beginning and work as expected.

Reply:
Well, I did something wrong I guess (by putting on the tag?) https://travis-ci.org/rust-lang/rust/builds/334956316

Reply:
Yeah, just remove the `[E0593]`.